tt_metal/impl/device/device.cpp

```
@@ -1753,6 +1753,8 @@ std::size_t v1::GetNumProgramCacheEntries(IDevice* device) { return device->num_
 tt::WorkExecutorMode Device::get_worker_mode() { return work_executor_->get_worker_mode(); }
 bool Device::is_worker_queue_empty() const { return work_executor_->worker_queue.empty(); }
 
+std::shared_ptr<distributed::MeshDevice> Device::get_mesh_device() { return mesh_device.lock(); }
```

is it valid to return nullptr here if `mesh_device` doesn't exist or should we just throw an exception?

---

Yes, it's valid if the device is not managed by any MeshDevice